eve/methods/post.py

```diff
@@ -271,8 +271,13 @@ def post_internal(resource, payl=None, skip_validation=False):
         for document in documents:
             # either return the custom ID_FIELD or the id returned by
             # data.insert().
-            id_ = document.get(id_field, ids.pop(0))
-            document[id_field] = id_
+            # documents updated or upserted through a bulk update will always have 
+            # the "_id" field set.
+            if "_id" in document:
+                document[id_field] = document['_id']
```

Comment:
As discussed, you can remove the code in "then" block. In case of update, document will be guaranteed to have "_id" field set.